addons/mail/static/src/models/notification_group/notification_group.js

```diff
@@ -67,6 +67,28 @@ function factory(dependencies) {
             return `${this.modelName}_${data.id}`;
         }
 
+        _computeDate() {
+            const groupDate = moment.max(this.notifications.map(notification => notification.message.date));
+            if (groupDate.isValid()) {
+                return groupDate;
+            }
+        }
+
+        _computeResId() {
+            const threadIds = this.notifications.map(notification => notification.message.originThread.id);
+            // keep res_id only if all notifications are for the same record
+            // set null if multiple records are present in the group
+            if (threadIds.every((threadId, index, array) => threadId === array[0])) {
+                return threadIds[0];
+            }
+            return null;
+
```

Comment:
This could more easily be replaced by using a Set for `threadIds`, and if the set has a size > 1 it means there are multiples.